src/Core/Domain/Product/SpecificPrice/Command/EditProductSpecificPriceCommand.php

```diff
@@ -0,0 +1,406 @@
+
+class EditProductSpecificPriceCommand
+{
+    /**
+     * //@todo: replace with Product/SpecificPriceId
+     *
+     * @var SpecificPriceId
+     */
+    private $specificPriceId;
+
+    /**
+     * @var Reduction|null
+     */
+    private $reduction;
+
+    /**
+     * @var bool|null
+     */
+    private $includesTax;
+
+    /**
+     * @var DecimalNumber|null
+     */
+    private $price;
+
+    /**
+     * @var int|null
+     */
+    private $fromQuantity;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce ShopGroupIdInterface (refer to example of ManufacturerIdInterface)
+     */
+    private $shopGroupId;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce ShopIdInterface (refer to example of ManufacturerIdInterface)
+     */
+    private $shopId;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce CombinationIdInterface (refer to example of ManufacturerIdInterface)
+     */
+    private $combinationId;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce CurrencyIdInterface (refer to example of ManufacturerIdInterface)
+     */
+    private $currencyId;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce CountryIdInterface (refer to example of ManufacturerIdInterface)
+     */
+    private $countryId;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce GroupIdInterface (refer to example of ManufacturerIdInterface). GroupId VO doesnt exist yet
+     */
+    private $groupId;
+
+    /**
+     * @var int|null
+     *
+     * @todo: introduce CustomerIdInterface (refer to example of ManufacturerIdInterface)
+     */
+    private $customerId;
+
+    /**
+     * @todo: its impossible to reset $from to NULL (database doesnt support null and the 0000-00-00 is actually invalid)
+     *      for that we could have some "DateRange" Value object which would contain ranges OR some "no-range" value.
+     *      Also this would require BC break - making datetime column nullable in specific_price table.
+     *
+     * @var DateTime|null
+     */
+    private $dateTimeFrom;
+
+    /**
+     * @var DateTime|null
+     */
+    private $dateTimeTo;
+
+    /**
+     * @param int $specificPriceId
+     */
+    public function __construct(int $specificPriceId)
+    {
+        $this->specificPriceId = new SpecificPriceId($specificPriceId);
+    }
+
+    /**
+     * @return SpecificPriceId
+     */
+    public function getSpecificPriceId(): SpecificPriceId
+    {
+        return $this->specificPriceId;
+    }
+
+    /**
+     * @return Reduction|null
+     */
+    public function getReduction(): ?Reduction
+    {
+        return $this->reduction;
+    }
+
+    /**
+     * @param string $reductionType
+     * @param float $reductionValue
+     *
+     * @todo: reduction value could also be a numeric string, because it is later converted to DecimalNumber.
+     *      (would require to refacto AddProductSpecificPriceCommand too)
```

Comment:
:up: it would mean BC break in AddProuctSpecificPriceCommand. is it worth it?

Reply:
It's better to use string values indeed to avoid float errors
I think it can be modified in the add command, especially since the page was still experimental I don't know if it can strictly be considered as a BC in this case 🤷‍♂️
Strictly speaking I guess it is but I'd say it's acceptable